### PR TITLE
fix(Core/Spells): Fixed calculating LoS checks for spells casted by g…

### LINKED ISSUE #12066: Unable to pick up several Battleground buffs.
https://github.com/chromiecraft/chromiecraft/issues/3456

### What client do you play on?

enUS

### Faction

- [X] Alliance
- [x] Horde

### Content Phase:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

Player is not able to pick up the buffs in both Horde huts and the Berserking Hut in Warsong Gulch and the buff located in the cart in Arathi Basin's Gold Mine and the Lumber Mill. The buff objects disappear without the buff being applied.

### Expected Blizzlike Behaviour

Buff applying

### Source

_No response_

### Steps to reproduce the problem

queue WSG
go to either Horde huts to pick up the buff
go to the Alliance Berserking hut to pick up the buff
queue AB
go to the Gold Mine or Lumber Mill and try to pick up the buffs

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/59e45c251eb53abb0ed4b37a57254cbdf415e69a

### Operating system

Win 7

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
…ameobjects.

Fixed #12066

## Issues Addressed:

- Closes #12066
- Closes https://github.com/chromiecraft/chromiecraft/issues/3456

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.debug bg`
queue WSG
go to either Horde huts to pick up the buff
go to the Alliance Berserking hut to pick up the buff

queue AB
go to the Gold Mine or Lumber Mill and try to pick up the buffs

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Tested with all the buffs, worked for all of them 👍 

Tried to pick them up through walls and couldn't.
The wagon buff in Gold Mine, however, could be picked up without jumping in the wagon, and just passing by it very close. But that's such a minor issue and could be merged as-is. Besides, it could even be blizzlike for all I know :p

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by acidmanifesto** [comment]
waiting for the compiling testing to be completed before we hit the merge button.

**Comment by acidmanifesto** [comment]
Thank you for the PR

**Pull request merged**
